sync/rpc/server.go

```diff
@@ -778,8 +785,8 @@ func (s *SyncAgentServer) SnapshotPurge(ctx context.Context, req *SnapshotPurgeR
 		var err error
 
 		defer func() {
-			if err != nil {
```

Comment:
Just do

```
if err != nil {
    s.PurgeStatus.Error = err
}
```

And you would need a lock for updating and reading PurgeStatus, since read/write can happen at the same time.